core/src/main/java/com/linecorp/armeria/server/AnnotatedHttpServiceMethod.java

```diff
@@ -521,6 +521,8 @@ private static String httpHeaderValue(Parameter entry, HttpRequest req) {
                 return (T) Float.valueOf(str);
             } else if (clazz == Double.TYPE) {
                 return (T) Double.valueOf(str);
+            } else if (clazz.isEnum()) {
+                return (T) Enum.valueOf(clazz.asSubclass(Enum.class), str);
```

Comment:
Could you replace the `catch (NumberFormatException e)` below with `catch (IllegalArgumentException e)` so we do not wrap an IAE with an IAE?